CADETProcess/optimization/surrogate.py

```diff
@@ -0,0 +1,119 @@
+import numpy as np
+
+from sklearn.gaussian_process import (
+    GaussianProcessRegressor, GaussianProcessClassifier)
+from sklearn.base import BaseEstimator
+
+from CADETProcess.optimization import (
+    Population, OptimizationProblem, OptimizationResults)
+
+class Surrogate:
+    def __init__(
+        self,
+        optimization_problem: OptimizationProblem,
+        population: Population
+    ):
+        self.optimization_problem = optimization_problem
+        self.surrogate_model_F: BaseEstimator = None
+        self.surrogate_model_G: BaseEstimator = None
+        self.surrogate_model_M: BaseEstimator = None
+        self.surrogate_model_CV: BaseEstimator = None
+        self.fit_gaussian_process(population)
+
+        # save a backup of bounds
+        self.lower_bounds_copy = optimization_problem.lower_bounds.copy()
+        self.upper_bounds_copy = optimization_problem.upper_bounds.copy()
+
+    def _reset_bounds_on_variables(self):
+        for var, lb, ub in zip(
+            self.optimization_problem.variables,
+            self.lower_bounds_copy,
+            self.upper_bounds_copy
+        ):
+            var.lb = lb
+            var.ub = ub
+
+    def fit_gaussian_process(self, population: Population):
```

Comment:
I'm wondering, maybe we should already pass a population in the constructor and provide an `update_model` function? This way we never encounter an un-initialized surrogate... But I guess it depends on where we put it in the architecture...

Reply:
already changed